dev/mock_source_data.py

```diff
@@ -0,0 +1,165 @@
+			"awsLambdaArn": "",
+		},
+	},
+	"dateFilter": "",
+	"hasStableIdentifiers": False
+}
+
+SOURCE_NAMES_TO_PARSERS = {
+	"argentina": "argentina-argentina-ingestor-locale2e",
+	"brazil_acre": "brazil_acre-acre-ingestor-locale2e",
+	"brazil_amapa": "brazil_amapa-amapa-ingestor-locale2e",
+	"brazil_ceara": "brazil_ceara-ceara-ingestor-locale2e",
+	"brazil_distrito_federal": "brazil_distrito_federal-distrito_federal-ingestor-locale2e",
+	"brazil_espirito_santo": "brazil_espirito_santo-espirito_santo-ingestor-locale2e",
+	"brazil_goias": "brazil_goias-goias-ingestor-locale2e",
+	"brazil_para": "brazil_para-para-ingestor-locale2e",
+	"brazil_paraiba": "brazil_paraiba-paraiba-ingestor-locale2e",
+	"brazil_rio_de_janeiro": "brazil_rio_de_janeiro-rio_de_janeiro-ingestor-locale2e",
+	"brazil_santa_catarina": "brazil_santa_catarina-santa_catarina-ingestor-locale2e",
+	"brazil_srag": "brazil_srag-srag-ingestor-locale2e",
+	"canada": "canada-canada-ingestor-locale2e",
+	"ch_zurich": "ch_zurich-zurich-ingestor-locale2e",
+	"colombia": "colombia-colombia-ingestor-locale2e",
+	"cuba": "cuba-cuba-ingestor-locale2e",
+	"czechia": "czechia-czechia-ingestor-locale2e",
+	"england": "england-england-ingestor-locale2e",
+	"estonia": "estonia-estonia-ingestor-locale2e",
+	"example": "example-example-ingestor-locale2e",
+	"germany": "germany-germany-ingestor-locale2e",
+	"hongkong": "hongkong-hongkong-ingestor-locale2e",
+	# TODO: add India (breaking due to two sample_data files)
+	"japan": "japan-japan-ingestor-locale2e",
+	"mexico": "mexico-mexico-ingestor-locale2e",
+	"new_zealand": "new_zealand-new_zealand-ingestor-locale2e",
+	"paraguay": "paraguay-paraguay-ingestor-locale2e",
+	"peru": "peru-peru-ingestor-locale2e",
+	"republic_of_korea": "republic_of_korea-republic_of_korea-ingestor-locale2e",
+	"riograndedosul": "riograndedosul-riograndedosul-ingestor-locale2e",
+	"saopaolo": "saopaolo-saopaolo-ingestor-locale2e",
+	"scotland": "scotland-scotland-ingestor-locale2e",
+	"south_africa": "south_africa-south_africa-ingestor-locale2e",
+	"taiwan": "taiwan-taiwan-ingestor-locale2e",
+	"thailand": "thailand-thailand-ingestor-locale2e",
+	"usa": "USA-USA-ingestor-locale2e",
+	"variants": "variants-variants-ingestor-locale2e",
+}
+
+SOURCE_IDS_TO_NAMES = {}
+
+UPLOAD_RESPONSE = {
+	"_id": "".join("0" for i in range(24))
+}
+
+
+app = Flask(__name__)
+
+
+class MockSourceData(object):
+	def __init__(self):
+		self.s3_client = boto3.client("s3", endpoint_url=LOCALSTACK_URL)
+		self.source_details = {}
+
+	def get_sample_data_file(self, source_id):
+		source_name = SOURCE_IDS_TO_NAMES.get(source_id)
+		if not source_name:
+			raise Exception(f"Missing source name for source id {source_id}")
+
+		response = self.s3_client.list_objects(
+			Bucket=BUCKET_NAME,
+			Prefix=source_name
+		)
+		objects = response.get("Contents", [])
+		if not objects:
+			raise Exception(f"No sample data files present for {source_name}")
+		obj = objects[0]
+		obj_key = obj.get("Key")
```

Comment:
Just a nit - you can use walrus := (assignment) operator to collapse assignment and value checking https://docs.python.org/3/reference/expressions.html#assignment-expressions

Reply:
I need to use the walrus more. In this block of code, I think it's more readable without them though:
```
		if objects := response.get("Contents", []):
                    obj = objects[0]
		    if obj_key := obj.get("Key"):
                        return obj_key
                    else:
                        raise Exception(f"No object key present for {source_name}")
		else:
		    raise Exception(f"No sample data files present for {source_name}")
		
```
Going forward I will try to use them more when the code follows a `if x exists then do something` pattern.